docs/design/embedded-agent-worker.md

```
@@ -0,0 +1,597 @@
+
+- `type` union gains `'embedded-agent'` (`:85`).
+- New nullable column `embedded_agent_id: string | null` (do NOT reuse `agent_id` — different registry namespace).
+- `pid` column (`:92-93`) is REUSED: the agent subprocess pid, `null` when not activated. This is what plugs embedded-agent workers into orphan reaping unchanged.
+
+Same migration as above adds the column. Update `VALID_WORKER_TYPES` (`mappers.ts:169`) and add branches:
+
+| Site | File | Change |
+|---|---|---|
+| `toWorkerRow` | `packages/server/src/database/mappers.ts:129-164` | embedded-agent branch: `pid` from worker, `embedded_agent_id`, `agent_id: null`, `base_commit: null` |
+| `toPersistedWorker` (DB->persisted) | `mappers.ts:179-222` | embedded-agent branch; throw `DataIntegrityError` if `embedded_agent_id` is null |
+| `PersistedWorker` union | `packages/server/src/services/persistence-service.ts:49-71` | add `PersistedEmbeddedAgentWorker { type: 'embedded-agent'; embeddedAgentId: string; pid: number | null }` |
```

markdownlint-cli2 (0.22.1)

[warning] Table column count
Expected: 3; Actual: 4; Too many cells, extra data will be missing

(MD056, table-column-count)